### ISSUE TITLE
(TSM) Suppress Recipe Master window

### ISSUE THREAD

**Issue by Gogo1951** [BrenoLudgero/Recipe_Master#9]
Auction, Trade, Bank, Vendor, Mailbox events should suppress Recipe Master, I think.

Or else it ends up adding clutter.

<img width="1966" alt="image" src="https://github.com/user-attachments/assets/667ce516-4833-4310-96f8-a89c955d70c6">

Pulled from the check I do in Open Sesame 
https://github.com/Gogo1951/Open-Sesame/blob/main/Open-Sesame_Vanilla.lua

```
-- Function to check if any specified windows or conditions are active
local function WindowOpen()
    -- Check for windows like Trade, Merchant, Bank, etc.
    return TradeFrame and TradeFrame:IsVisible() or MerchantFrame and MerchantFrame:IsVisible() or
        BankFrame and BankFrame:IsVisible() or
        (ContainerFrame1 and ContainerFrame1:IsVisible()) or -- Backpack
        (ContainerFrame2 and ContainerFrame2:IsVisible()) or -- First bag
        (ContainerFrame3 and ContainerFrame3:IsVisible()) or -- Second bag
        (ContainerFrame4 and ContainerFrame4:IsVisible()) or -- Third bag
        (ContainerFrame5 and ContainerFrame5:IsVisible()) or -- Fourth bag
        (GossipFrame and GossipFrame:IsVisible()) or
        (AuctionFrame and AuctionFrame:IsVisible()) or
        (ReagentBankFrame and ReagentBankFrame:IsVisible()) or
        (CraftFrame and CraftFrame:IsVisible()) or
        (TradeSkillFrame and TradeSkillFrame:IsVisible()) or
        LootFrame and LootFrame:IsVisible() or
        MailFrame and MailFrame:IsVisible()
end
```

**Comment by BrenoLudgero** [comment]
Hello, @Gogo1951.

You can now minimize the Recipe Master window at will while TradeSkillMaster is enabled!

Please let me know if this feature gives you the result you were looking for.

And thank you for the suggestion!

**Comment by BrenoLudgero** [comment]
This fix has been reverted in commit ddc84009843d69b629bc774a1574624909ea0ae2 because it introduced too many bugs.

**Comment by BrenoLudgero** [comment]
Commit 56f32f5 added the ability to minimize the window to a moveable button when TSM is enabled.
